assets/js/components/charts/Forecasts.jsx

```diff
@@ -118,16 +118,58 @@ export default class Forecasts extends Component<Props> {
       .x((d) => x(d.time))
       .y((d) => y(d.value))
 
+    var tooltip = d3
+      .select("body")
+      .append("div")
+      .classed("forecast-tooltip", true)
+      .style("visibility", "hidden")
+
+    for (var i = 0; i <= data.length; i++) {
+      if (data[i]) {
```

Comment:
I had to check here for `data[i]` otherwise `data[i].values` would raise an error in the line below, meaning that `data.length` is > 0 but has not defined `values`, which for me was a not expected behavior. Same with the if over `data[i].values[j]` two lines below.